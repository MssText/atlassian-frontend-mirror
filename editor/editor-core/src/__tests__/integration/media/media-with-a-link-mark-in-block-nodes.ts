import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import { editable, getDocFromElement, fullpage } from '../_helpers';
import {
  mountEditor,
  goToEditorTestingExample,
} from '../../__helpers/testing-example-helpers';
import adfMediaSingleWithLinkInALayoutColumn from './_fixtures_/media-single-with-a-link-mark-in-a-layout-column.adf.json';
import adfMediaSingleWithLinkInAListItem from './_fixtures_/media-single-with-a-link-mark-in-a-list-item.adf.json';
import adfMediaSingleWithLinkInANestedExpand from './_fixtures_/media-single-with-a-link-mark-in-a-nested-expand.adf.json';
import adfMediaSingleWithLinkInATable from './_fixtures_/media-single-with-a-link-mark-in-a-table.adf.json';
import adfMediaSingleWithLinkInAnExpand from './_fixtures_/media-single-with-a-link-mark-in-an-expand.adf.json';

async function loadADF(client: any, adf: object): Promise<any> {
  const page = await goToEditorTestingExample(client);

  await mountEditor(page, {
    appearance: fullpage.appearance,
    defaultValue: adf,
    media: {
      allowMediaSingle: true,
    },
  });

  return await page.$eval(editable, getDocFromElement);
}

BrowserTestCase(
  'should load the adf with media single inside a layout column',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const doc = await loadADF(client, adfMediaSingleWithLinkInALayoutColumn);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'should load the adf with media single inside a table cell',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const doc = await loadADF(client, adfMediaSingleWithLinkInATable);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'should load the adf with media single inside a list item',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const doc = await loadADF(client, adfMediaSingleWithLinkInAListItem);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'should load the adf with media single inside an expand',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const doc = await loadADF(client, adfMediaSingleWithLinkInAnExpand);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);

BrowserTestCase(
  'should load the adf with media single inside a nested expand',
  { skip: ['edge'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: adfMediaSingleWithLinkInANestedExpand,
      media: {
        allowMediaSingle: true,
      },
    });

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
