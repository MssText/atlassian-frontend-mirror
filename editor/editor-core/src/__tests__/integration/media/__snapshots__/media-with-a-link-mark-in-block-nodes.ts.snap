// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should load the adf with media single inside a layout column 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "originalValue": Object {
          "content": Array [
            Object {
              "attrs": Object {
                "width": 50,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "layout": "center",
                  },
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "collection": "MediaServicesSample",
                        "height": 683,
                        "id": "f3522853-855b-471b-869c-6d9a8c424eea",
                        "type": "file",
                        "width": 1024,
                      },
                      "type": "media",
                    },
                  ],
                  "marks": Array [
                    Object {
                      "attrs": Object {
                        "href": "https://gnu.org",
                      },
                      "type": "link",
                    },
                  ],
                  "type": "mediaSingle",
                },
              ],
              "type": "layoutColumn",
            },
            Object {
              "attrs": Object {
                "width": 50,
              },
              "content": Array [
                Object {
                  "content": Array [],
                  "type": "paragraph",
                },
              ],
              "type": "layoutColumn",
            },
          ],
          "type": "layoutSection",
        },
      },
      "type": "unsupportedBlock",
    },
  ],
  "type": "doc",
}
`;

exports[`should load the adf with media single inside a list item 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Um",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Dois",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "attrs": Object {
                "layout": "center",
                "width": null,
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "__contextId": "DUMMY-OBJECT-ID",
                    "__displayType": null,
                    "__external": false,
                    "__fileMimeType": null,
                    "__fileName": "example.png",
                    "__fileSize": null,
                    "alt": null,
                    "collection": "MediaServicesSample",
                    "height": 381,
                    "id": "1234-5678-abcd-efgh",
                    "occurrenceKey": null,
                    "type": "file",
                    "url": null,
                    "width": 500,
                  },
                  "type": "media",
                },
              ],
              "marks": Array [
                Object {
                  "attrs": Object {
                    "__confluenceMetadata": null,
                    "href": "https://gnu.org",
                  },
                  "type": "link",
                },
              ],
              "type": "mediaSingle",
            },
            Object {
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
        Object {
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "text": "Tres",
                  "type": "text",
                },
              ],
              "type": "paragraph",
            },
          ],
          "type": "listItem",
        },
      ],
      "type": "bulletList",
    },
  ],
  "type": "doc",
}
`;

exports[`should load the adf with media single inside a nested expand 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "originalValue": Object {
          "attrs": Object {
            "isNumberColumnEnabled": false,
            "layout": "default",
          },
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "attrs": Object {},
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "title": "",
                      },
                      "content": Array [
                        Object {
                          "attrs": Object {
                            "layout": "center",
                          },
                          "content": Array [
                            Object {
                              "attrs": Object {
                                "collection": "MediaServicesSample",
                                "height": 381,
                                "id": "4b4bbf71-4ff5-4827-8b48-ac9e4e15cb68",
                                "type": "file",
                                "width": 500,
                              },
                              "type": "media",
                            },
                          ],
                          "marks": Array [
                            Object {
                              "attrs": Object {
                                "href": "https://gnu.org",
                              },
                              "type": "link",
                            },
                          ],
                          "type": "mediaSingle",
                        },
                        Object {
                          "content": Array [],
                          "type": "paragraph",
                        },
                      ],
                      "type": "nestedExpand",
                    },
                  ],
                  "type": "tableCell",
                },
              ],
              "type": "tableRow",
            },
          ],
          "type": "table",
        },
      },
      "type": "unsupportedBlock",
    },
  ],
  "type": "doc",
}
`;

exports[`should load the adf with media single inside a table cell 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "originalValue": Object {
          "attrs": Object {
            "isNumberColumnEnabled": false,
            "layout": "default",
          },
          "content": Array [
            Object {
              "content": Array [
                Object {
                  "attrs": Object {},
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "layout": "center",
                      },
                      "content": Array [
                        Object {
                          "attrs": Object {
                            "collection": "MediaServicesSample",
                            "height": 381,
                            "id": "6dca16ef-0e85-4e85-89be-f780ac7929ff",
                            "type": "file",
                            "width": 500,
                          },
                          "type": "media",
                        },
                      ],
                      "marks": Array [
                        Object {
                          "attrs": Object {
                            "href": "https://gnu.org",
                          },
                          "type": "link",
                        },
                      ],
                      "type": "mediaSingle",
                    },
                    Object {
                      "content": Array [],
                      "type": "paragraph",
                    },
                  ],
                  "type": "tableCell",
                },
              ],
              "type": "tableRow",
            },
          ],
          "type": "table",
        },
      },
      "type": "unsupportedBlock",
    },
  ],
  "type": "doc",
}
`;

exports[`should load the adf with media single inside an expand 1`] = `
Object {
  "content": Array [
    Object {
      "attrs": Object {
        "originalValue": Object {
          "attrs": Object {
            "title": "",
          },
          "content": Array [
            Object {
              "attrs": Object {
                "layout": "center",
              },
              "content": Array [
                Object {
                  "attrs": Object {
                    "collection": "MediaServicesSample",
                    "height": 381,
                    "id": "d70b7150-f3a5-4018-abee-cfb669f6867d",
                    "type": "file",
                    "width": 500,
                  },
                  "type": "media",
                },
              ],
              "marks": Array [
                Object {
                  "attrs": Object {
                    "href": "https://gnu.org",
                  },
                  "type": "link",
                },
              ],
              "type": "mediaSingle",
            },
            Object {
              "content": Array [],
              "type": "paragraph",
            },
          ],
          "type": "expand",
        },
      },
      "type": "unsupportedBlock",
    },
  ],
  "type": "doc",
}
`;
